torch/utils/_python_dispatch.py

```diff
@@ -164,3 +164,130 @@ def transform_subclass(t, callback):
     for attr in attrs:
         transformed_tensors_dict[attr] = callback(attr, getattr(t, attr))
     return type(t).__tensor_unflatten__(transformed_tensors_dict, ctx)
+
+def _correct_storage_aliasing(func, args, outs):
+    assert isinstance(func, torch._ops.OpOverload)
+    assert isinstance(args, tuple)
+    assert isinstance(outs, (list, tuple))
+    flat_outs, _ = torch.utils._pytree.tree_flatten(outs)
+    for x in flat_outs:
+        if isinstance(x, torch.Tensor):
+            assert hasattr(x, '__torch_dispatch__')
+
+    def alias_storage(arg, ret):
+        # This is hopefully a reasonable assert:
+        # subclasses that rely on this API for output aliasing
+        # should always return wrapper tensor subclasses for us to manually alias.
+        # in theory if a subclass that needs this API wants to sometimes return
+        # plain tensors, we could remove the assert and just not perform the aliasing,
+        # but it seems safer to learn more about this case first.
+        if is_traceable_wrapper_subclass(arg) or is_traceable_wrapper_subclass(ret):
+            assert type(arg) == type(ret), f"""Called {str(func)} with input of type {type(arg)}
+and output of type {type(ret)}. But expected types to match."""
+        # Need to run under no_dispatch, because we explicitly do **not**
+        # want our subclass to intercept the set_() call.
+        # instead, our subclass should directly have its storage swapped out.
+        with torch.utils._mode_utils.no_dispatch():
+            # directly calling this overload, and passing ret.shape, because we **explicitly**
+            # don't want to reset the sizes on ret, if the storage implies a size change.
+            # Why?
+            # The purpose of this API is *not* to change the size/strides of our output- we assume it's already correct.
+            # We just want to "fix up" the storage aliasing, without modifying or output's metadata.
+            # Example: out = inp.expand(inp.shape[0], inp.shape[0])
+            #     This requires swapping the storage of out to be the same as inp,
+            #     but we do *not* want it to change the sizes/strides that were compute for out.
+            meta_in_tls = torch._C._meta_in_tls_dispatch_include()
+            # TODO: make this a context manager
+            torch._C._set_meta_in_tls_dispatch_include(True)
+            try:
+                torch.ops.aten.set_.source_Storage_storage_offset(ret, arg.untyped_storage(), 0, ret.shape)
```

Comment:
Do you really want the offset to always be zero?

Reply:
sorry thanks.... `ret.storage_offset()`